src/libraries/System.IO.Pipes/src/System/IO/Pipes/NamedPipeServerStream.Windows.cs

```diff
@@ -141,6 +141,7 @@ internal override void TryToReuse(PipeValueTaskSource source)
 
                 if (handle.IsInvalid)
                 {
+                    handle.Dispose();
                     throw Win32Marshal.GetExceptionForLastWin32Error();
```

Comment:
Do we need to capture the exception prior to disposing, like in the many other earlier examples?

Reply:
Yup, thanks.